pkg/cmd/commands.go

```diff
@@ -180,3 +205,40 @@ func findAbsPath(baseFilePath, path string) string {
 	}
 	return filepath.Join(filepath.Dir(baseFilePath), path)
 }
+
+func validate(p []byte) {
+	var speco interface{}
+	err := yaml.Unmarshal(p, &speco)
+	if err != nil {
+		fmt.Printf("Error with Unmarhsalling")
+	}
+	body := convertToStringKeys(speco)
+	s := body.(map[string]interface{})
+	loader := gojsonschema.NewGoLoader(body)
+	if s["controller"] == nil {
+		s["controller"] = "deployment"
+	}
+
+	var schema gojsonschema.JSONLoader
+	if s["controller"] == "deployment" {
+		schema = gojsonschema.NewStringLoader(validation.DeploymentspecmodJson)
+	}
+	if s["controller"] == "job" {
+		schema = gojsonschema.NewStringLoader(validation.JobspecmodJson)
+	}
+
+	gojsonschema.FormatCheckers.Add("int64", ValidFormat{})
+	gojsonschema.FormatCheckers.Add("byte", ValidFormat{})
+	gojsonschema.FormatCheckers.Add("int32", ValidFormat{})
+	gojsonschema.FormatCheckers.Add("int-or-string", ValidFormat{})
+	result, err := gojsonschema.Validate(schema, loader)
+	if !result.Valid() {
+		fmt.Printf("The document is not valid. see errors :\n")
+		for _, err := range result.Errors() {
+			// Err implements the ResultError interface
+			fmt.Printf("- %s\n", err)
+			os.Exit(-1)
+		}
+	}
+
+}
```

Comment:
Although WIP, entire code here needs to be spaced out more / added comments in order to understand what's happening :+1:

Reply:
okay

Reply:
@cdrage , added comments now, I hope it will help now